modules/markdown/markdown.go

```diff
@@ -208,22 +210,30 @@ func cutoutVerbosePrefix(prefix string) string {
 // RenderIssueIndexPattern renders issue indexes to corresponding links.
 func RenderIssueIndexPattern(rawBytes []byte, urlPrefix string, metas map[string]string) []byte {
 	urlPrefix = cutoutVerbosePrefix(urlPrefix)
-	ms := IssueIndexPattern.FindAll(rawBytes, -1)
+
+	pattern := IssueNumericPattern
+	if metas["style"] == "alphanumeric" {
+		pattern = IssueAlphanumericPattern
+	}
+
+	ms := pattern.FindAll(rawBytes, -1)
 	for _, m := range ms {
-		var space string
-		if m[0] != '#' {
-			space = string(m[0])
-			m = m[1:]
+		if m[0] == ' ' || m[0] == '('  {
+			m = m[1:] // ignore leading space or opening parentheses
 		}
+		var link string
 		if metas == nil {
-			rawBytes = bytes.Replace(rawBytes, m, []byte(fmt.Sprintf(`%s<a href="%s/issues/%s">%s</a>`,
-				space, urlPrefix, m[1:], m)), 1)
+			link = fmt.Sprintf(`<a href="%s/issues/%s">%s</a>`, urlPrefix, m[1:], m)
 		} else {
 			// Support for external issue tracker
-			metas["index"] = string(m[1:])
-			rawBytes = bytes.Replace(rawBytes, m, []byte(fmt.Sprintf(`%s<a href="%s">%s</a>`,
-				space, com.Expand(metas["format"], metas), m)), 1)
+			if metas["style"] == "numeric" {
```

Comment:
I think here we should check `"alphanumeric"` and leave `else` for `"numeric"`, because that's the default one.